pkg/builder/in_memory_build_queue.go

```diff
@@ -311,14 +313,28 @@ func (bq *InMemoryBuildQueue) Execute(in *remoteexecution.ExecuteRequest, out re
 			QueuedTimestamp: bq.getCurrentTime(),
 		}
 
-		span := trace.FromContext(ctx)
-		if span != nil {
-			desiredState.TraceContext = propagation.Binary(span.SpanContext())
-		}
+		// This is close enough to
+		// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+		// to use their span format.
+		//
+		// Note that ctx here is local to producing the injection, we
+		// don't otherwise use it.
+		ctx, span := bq.tracer.Start(ctx, "bb_worker send",
+			trace.WithSpanKind(trace.SpanKindProducer),
+			trace.WithAttributes(
+				label.String("messaging.system", "bb_scheduler"),
+				label.String("messaging.destination", instanceName.String()),
```

Comment:
Picking the instance name here seems a bit arbitrary. Wouldn't it make more sense to pick the pair of [instance name, platform]? That identifies queues right now.

This code is also largely repeated in the scheduler and the worker. Could it be factored out?

Reply:
It can be partially factored out; now done.

I've taken a stab at working platform into this, but it's a bit of a mess. Notably bb_worker never validates its platform config is valid. I've stuffed it in to highlight the problem, let me know if you have any good ideas what to do with this.

Reply:
This is the major outstanding puzzle - we still need to figure out how best to solve this. I'll make another pass later this week and see if I have any better ideas.